pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -69,10 +72,46 @@ func nodeSelectorMatchesThisNode(cl client.Client, eventObject runtime.Object) b
 	return matches(policyNodeSelector, node.ObjectMeta.Labels)
 }
 
+func initializeEnactment(client client.Client, policy *nmstatev1alpha1.NodeNetworkConfigurationPolicy) (*nmstatev1alpha1.NodeNetworkConfigurationEnactment, error) {
+
+	node := corev1.Node{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: nodeName}, &node)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("cannot get node %s", nodeName))
+	}
+
+	ownerRefList := []metav1.OwnerReference{
+		{Name: policy.Name, Kind: "NodeNetworkConfigurationPolicy", APIVersion: "v1alpha1", UID: policy.UID},
+		{Name: nodeName, Kind: "Node", APIVersion: "v1", UID: node.UID},
+	}
+
+	enactment := nmstatev1alpha1.NodeNetworkConfigurationEnactment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            nmstatev1alpha1.EnactmentKey(node.Name, policy.Name).Name,
+			OwnerReferences: ownerRefList,
+		},
+	}
+
+	err = client.Create(context.TODO(), &enactment)
+	if err != nil {
+		return nil, fmt.Errorf("error creating NodeNetworkConfigurationEnactment: %v, %+v", err, enactment)
+	}
+
+	return &enactment, nil
+}
+
 func forThisNodePredicate(cl client.Client) predicate.Funcs {
 	return predicate.Funcs{
 		CreateFunc: func(createEvent event.CreateEvent) bool {
-			return nodeSelectorMatchesThisNode(cl, createEvent.Object)
+			if nodeSelectorMatchesThisNode(cl, createEvent.Object) {
+				policy := createEvent.Object.(*nmstatev1alpha1.NodeNetworkConfigurationPolicy)
+				_, err := initializeEnactment(cl, policy)
```

Comment:
We should not do any side effects in predicate. The initialization have to happen inside the reconcile function. If the NNCE does not exist yet, create it.

Reply:
Then how we will state Matching state to False ?

Reply:
We need to move the Matching check to Reconcile function if we want to act (read "create objects on API") based on it.

Reply:
Moved the initialize to Reconcile, well move the node check at Reconcile at Matching state PR.